src/main/java/Duke.java

```diff
@@ -1,10 +1,171 @@
+                        t.markAsDone();
+                        System.out.println("Nice! Congratulations for completing this task:\n" + t);
+                        updateTaskData(taskDataPath, tasks);
+                        break;
+                    case TODO:
+                        String todoStuff = st.nextToken("").trim();
+//                        if (todoStuff.equals("")) {
+//                            throw new IndexOutOfBoundsException("OOPS!!! The description of a todo cannot be empty.");
+//                        }
+                        Todo todo = new Todo(todoStuff);
+                        tasks.add(todo);
+                        String todoMsg = "Got it. I've added this task:\n" + todo
+                                + String.format("\nnow you have %d tasks in the list.", tasks.size());
+                        System.out.println(todoMsg);
+                        updateTaskData(taskDataPath, tasks);
+                        break;
+                    case DEADLINE:
+                        String ddlStuff = st.nextToken("/").trim();
+                        String ddlTime = st.nextToken("/").trim().substring(3);
+                        Deadline ddl = new Deadline(ddlStuff, ddlTime);
+                        tasks.add(ddl);
+                        String ddlMsg = "Got it. I've added this task:\n" + ddl
+                                + String.format("\nnow you have %d tasks in the list.", tasks.size());
+                        System.out.println(ddlMsg);
+                        updateTaskData(taskDataPath, tasks);
+                        break;
+                    case EVENT:
+                        String eventStuff = st.nextToken("/").trim();
+                        String eventTime = st.nextToken("/").trim().substring(3);
+                        Event event = new Event(eventStuff, eventTime);
+                        tasks.add(event);
+                        String eventMsg = "Got it. I've added this task:\n" + event
+                                + String.format("\nnow you have %d tasks in the list.", tasks.size());
+                        System.out.println(eventMsg);
+                        updateTaskData(taskDataPath, tasks);
+                        break;
+                    case DELETE:
+                        int num = Integer.parseInt(st.nextToken());
+                        Task del = tasks.get(num - 1);
+                        tasks.remove(del);
+                        String deleteMsg = "Noted. I've removed this task:\n" + del
+                                + String.format("\nnow you have %d tasks in the list.", tasks.size());
+                        System.out.println(deleteMsg);
+                        updateTaskData(taskDataPath, tasks);
+                        break;
+                    default:
+                        throw new IllegalArgumentException("OOPS!!! I'm sorry, but I don't know what that means :-(");
+                }
```

Comment:
Good use of enum to categorise the commands!
However, Duke.java seems a bit too cluttered.
Maybe you can consider creating more OOP classes to organise the different functionalities?